//-*-mode:protobuf; mode:font-lock;-*-

/*

   calin/proto/simulations/pmt.proto -- Stephen Fegan -- 2016-03-21

   Protobufs for defining PMT simulations

   Copyright 2016, Stephen Fegan <sfegan@llr.in2p3.fr>
   LLR, Ecole polytechnique, CNRS/IN2P3, Universite Paris-Saclay

   This file is part of "calin"

   "calin" is free software: you can redistribute it and/or modify it
   under the terms of the GNU General Public License version 2 or
   later, as published by the Free Software Foundation.

   "calin" is distributed in the hope that it will be useful, but
   WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
   General Public License for more details.

*/

syntax = "proto3";

import "calin.proto";
import "common_types.proto";

package calin.ix.simulation.pmt;

message PMTSimStageConfig {
  double gain_mean                                         = 1 [
    (CFO).desc = "Gain of stage." ];
  double gain_rms_frac                                     = 2 [
    (CFO).desc = "Additional non-Poisson RMS of stage as fraction of mean." ];
  double prob_skip                                         = 3 [
    (CFO).desc = "Probability of skipping stage." ];
};

message PMTSimConfig {
  repeated PMTSimStageConfig stage                         = 1 [
    (CFO).desc = "Configuration values for all stages." ];
  bool suppress_zero                                       = 2 [
    (CFO).desc = "Suppress events with zero charge." ];
  bool signal_in_pe                                        = 3 [
    (CFO).desc = "Rescale output into PE by dividing by mean gain." ];
};

message PMTSimAbbreviatedConfig {
  uint32 num_stage                                         = 1 [
    (CFO).desc = "Number of gain stages (dynodes)." ];
  double total_gain                                        = 2 [
    (CFO).desc = "Total gain of the PMT, including zero-charge events." ];
  double stage_1_gain                                       = 3 [
    (CFO).desc = "Mean gain of stage 1." ];
  double stage_1_gain_rms_frac                              = 4 [
    (CFO).desc = "Additional non-Poisson RMS of stage 1 as fraction of mean." ];
  double stage_1_prob_skip                                  = 5 [
    (CFO).desc = "Probability of skipping stage 1." ];
  double stage_n_gain_rms_frac                              = 6 [
    (CFO).desc = "Additional non-Poisson RMS of stage n>1 as fraction of mean." ];
  double stage_n_prob_skip                                  = 7 [
    (CFO).desc = "Probability of skipping stage n>1." ];
  bool suppress_zero                                       = 8 [
    (CFO).desc = "Suppress events with zero charge." ];
  bool signal_in_pe                                        = 9 [
    (CFO).desc = "Rescale output into PE by dividing by mean gain." ];
};
